linkerd/app/outbound/src/http/peer_proxy_errors.rs

```diff
@@ -41,146 +37,89 @@ where
     }
 }
 
-impl<S, A, B> Service<Request<A>> for PeerProxyErrors<S>
+impl<S, A, B> svc::Service<http::Request<A>> for PeerProxyErrors<S>
 where
-    S: Service<Request<A>, Response = Response<B>>,
+    S: svc::Service<http::Request<A>, Response = http::Response<B>>,
     S::Response: Send,
     S::Future: Send + 'static,
-    A: Send + 'static,
     B: Send + 'static,
 {
     type Response = S::Response;
     type Error = S::Error;
     type Future = Pin<Box<dyn Future<Output = Result<S::Response, S::Error>> + Send + 'static>>;
 
+    #[inline]
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         self.inner.poll_ready(cx)
     }
 
-    fn call(&mut self, req: Request<A>) -> Self::Future {
+    fn call(&mut self, req: http::Request<A>) -> Self::Future {
         let client = req.extensions().get::<ClientHandle>().cloned();
         debug_assert!(client.is_some(), "Missing client handle");
-        let response = self.inner.call(req);
-        Box::pin(async move {
-            let response = response.await?;
-            if let Some(message) = response.headers().get(L5D_PROXY_ERROR) {
-                tracing::info!(?message, "response contained `{}` header", L5D_PROXY_ERROR);
+
+        Box::pin(self.inner.call(req).map_ok(move |rsp| {
+            if let Some(msg) = rsp.headers().get(L5D_PROXY_ERROR) {
+                tracing::debug!(header = %L5D_PROXY_ERROR, ?msg);
+
                 // Gracefully teardown the accepted connection.
                 if let Some(ClientHandle { close, .. }) = client {
                     tracing::trace!("connection closed");
                     close.close();
                 }
             }
-            Ok(response)
-        })
+
+            rsp
```

Comment:
Generally, we use req/rsp for request and response. Especially when the variable is short-lived. Basically: variables with short scopes get shorter names. Important variables with large scopes get longer names.